src/main/scala/mesosphere/marathon/state/PortDefinition.scala

```diff
@@ -14,7 +14,7 @@ case class PortDefinition(
 
 object PortDefinition {
   implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
-    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
+    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
```

Comment:
huh? I'm talking about the upcoming pods RAML, suggesting that protocols
could be a string[] instead of a string. we shouldn't enforce an enum on
protocols since mesos doesn't (what would be the point of marathon limiting
the use of the field when mesos does not?)

On Thu, Sep 1, 2016 at 1:05 PM, Jason Gilanfarr <notifications@github.com>
wrote:

> In src/main/scala/mesosphere/marathon/state/PortDefinition.scala
> <https://github.com/mesosphere/marathon/pull/4284#discussion_r77213889>:
>
> > @@ -14,7 +14,7 @@ case class PortDefinition(
> >
> >  object PortDefinition {
> >    implicit val portDefinitionValidator = validator[PortDefinition] { portDefinition =>
> > -    portDefinition.protocol is oneOf(DiscoveryInfo.Port.AllowedProtocols)
> > +    portDefinition.protocol.split(',').toIterable is every(oneOf(DiscoveryInfo.Port.AllowedProtocols))
>
> If we changed it to an array, we'd potentially break compatibility unless
> we make it a union of either enum or array of enum.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mesosphere/marathon/pull/4284/files/a1c71b58309775a3ca78fc02af2fe987bc5ee603#r77213889>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACPVLGFGxeAv-yvF94dbv97A6s__Qrm0ks5qlwXlgaJpZM4Jysrn>
> .
>